src/stream/stream.rs

```diff
@@ -111,6 +112,68 @@ pub trait Stream {
             remaining: n,
         }
     }
+
+    /// Tests if every element of the stream matches a predicate.
+    ///
+    /// `all()` takes a closure that returns `true` or `false`. It applies
+    /// this closure to each element of the stream, and if they all return
+    /// `true`, then so does `all()`. If any of them return `false`, it
+    /// returns `false`.
+    ///
+    /// `all()` is short-circuiting; in other words, it will stop processing
+    /// as soon as it finds a `false`, given that no matter what else happens,
+    /// the result will also be `false`.
+    ///
+    /// An empty stream returns `true`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// # fn main() { async_std::task::block_on(async {
+    /// #
+    /// use async_std::prelude::*;
+    /// use async_std::stream;
+    ///
+    /// let mut s = stream::repeat::<u32>(42).take(3);
+    /// assert!(s.all(|x| x ==  42).await);
+    ///
+    /// #
+    /// # }) }
+    /// ```
+    ///
+    /// Empty stream:
+    ///
+    /// ```
+    /// # fn main() { async_std::task::block_on(async {
+    /// #
+    /// use async_std::prelude::*;
+    /// use async_std::stream;
+    ///
+    /// let mut s = stream::empty::<u32>();
+    /// assert!(s.all(|_| false).await);
+    ///
+    /// #
+    /// # }) }
+    /// ```
+    /// Stopping at the first `false`:
+    ///
+    /// TODO: add example here
+    /// TODO: add more examples
```

Comment:
I think it's fine to just have the examples you already made!

Reply:
ok